src/adapter/src/catalog/builtin.rs

```diff
@@ -1698,8 +1713,15 @@ pub const PG_ATTRIBUTE: BuiltinView = BuiltinView {
     FALSE as attisdropped,
     -- MZ doesn't support COLLATE so attcollation is filled with 0
     0::pg_catalog.oid as attcollation
-FROM mz_catalog.mz_objects
-JOIN mz_catalog.mz_columns ON mz_objects.id = mz_columns.id
+FROM (
+    -- pg_attribute catalogs columns on relations and indexes
+    SELECT id, oid, schema_id, name, type FROM mz_catalog.mz_relations
+    UNION
+        SELECT mz_indexes.id, mz_indexes.oid, schema_id, mz_indexes.name, 'index'
```

Comment:
Nits: `mz_relations.schema_id` makes it clearer where `schema_id` is coming from. `'index' AS type` makes it a bit clearer where `index` is going.